examples/source_separation/conv_tasnet/trainer.py

```diff
@@ -0,0 +1,77 @@
+import time
+
+import torch
+import torch.distributed as dist
+
+from utils import dist_utils
+
+_LG = dist_utils.getLogger(__name__)
+
+
+class Trainer:
+    def __init__(
+        self, model, optimizer, train_loader, eval_loader, criterion, grad_clip, device,
+    ):
+        self.model = model
+        self.optimizer = optimizer
+        self.train_loader = train_loader
+        self.eval_loader = eval_loader
+        self.criterion = criterion
+        self.grad_clip = grad_clip
+        self.device = device
+
+    def train_one_epoch(self):
+        num_batches = len(self.train_loader)
+        last_log_time = 0
+        last_log_progress = 0
+        t0 = time.monotonic()
+        for i, (input, reference) in enumerate(self.train_loader, start=1):
```

Comment:
nit: I found it useful to have a generator a la `tqdm` to handle logging / timing for me, see https://github.com/facebookresearch/detr/blob/5e66b4cd15b2b182da347103dd16578d28b49d69/util/misc.py#L193-L245
although it might just be an overkill and might not be needed here

Reply:
I have an exact opposite idea. `tqdm` messes up the log when running in background, and adding more code do disable it goes against of writing a simple example. Also the interactive progress bar is not useful in the first place to me as I never launch training process in foreground except debugging.